src/rebar_utils.erl

```diff
@@ -717,7 +717,10 @@ update_code(Paths, Opts) ->
                                   code:add_patha(Path),
                                   case lists:member(soft_purge, Opts) of
                                       true  ->
-                                          [begin code:soft_purge(M), code:delete(M) end || M <- Modules];
+                                            %% only soft purge modules that were not
+                                            %% cover compiled, otherwise coverage is lost
+                                          [begin code:soft_purge(M), code:delete(M) end ||
+                                                M <- Modules, cover:is_compiled(M) =:= false];
```

Comment:
The only thing I'm wondering about is what would happen after cover compilation is done; I think currently our test handlers don't clean up cover-compiled modules and so we kill the loading functions, but not too sure this will be problematic this time around.